src/shared/components/navbar.tsx

```diff
@@ -1,13 +1,22 @@
-import { Component } from "inferno";
+import { Component, ChangeEvent } from "inferno";
 import { Link } from "inferno-router";
 import { LinkLine } from "./link-line";
 import { Icon } from "./icon";
+import { getLanguageName, i18n } from "../i18next";
 
 export class Navbar extends Component<any, any> {
   constructor(props: any, context: any) {
     super(props, context);
   }
 
+  handleLanguageChange(event: ChangeEvent<HTMLSelectElement>) {
+    location.href = `/?lang=${event.target.value}`;
```

Comment:
I don't think we have any params-based things here, might just require setting the language cookie then reloading.

Reply:
That will also work, but sometimes its useful to link people to a specific language version of the site, and thats much easier if you can just use the selection and then copy the url.